react/OrderShipping.tsx

```diff
@@ -1,31 +1,195 @@
-import React, { createContext, ReactNode, useContext } from 'react'
-
+import React, { createContext, ReactNode, useContext, useCallback } from 'react'
+import { compose, graphql } from 'react-apollo'
+import { useOrderQueue } from 'vtex.order-manager/OrderQueue'
 import { useOrderForm } from 'vtex.order-manager/OrderForm'
 
+import {
+  estimateShipping as EstimateShipping,
+  selectDeliveryOption as SelectDeliveryOption,
+} from 'vtex.checkout-resources/Mutations'
+
 interface Context {
   countries: string[]
+  selectedAddress: CheckoutAddress
+  insertAddress: (address: CheckoutAddress) => void
+  deliveryOptions: DeliveryOption[]
+  selectDeliveryOption: (option: string) => void
 }
 
 interface OrderShippingProps {
   children: ReactNode
+  EstimateShipping: any
+  SelectDeliveryOption: any
 }
 
 const OrderShippingContext = createContext<Context | undefined>(undefined)
 
-export const OrderShippingProvider = ({ children }: OrderShippingProps) => {
-  const {
-    orderForm: {
-      shipping: { countries },
-    },
-  } = useOrderForm()
-
-  return (
-    <OrderShippingContext.Provider value={{ countries }}>
-      {children}
-    </OrderShippingContext.Provider>
-  )
+const shippingId = 'Shipping'
+const TASK_CANCELLED = 'TASK_CANCELLED'
+
+const changeSelectedDeliveryOption = (
+  deliveryOptions: DeliveryOption[],
+  selectedDeliveryOptionId: string
+) => {
+  return deliveryOptions.map(option => {
+    if (option.isSelected) {
+      option.isSelected = false
+    } else if (option.id === selectedDeliveryOptionId) {
+      option.isSelected = true
+    }
+
+    return option
+  })
+}
+
+const updateShipping = (totalizers: Totalizer[], newShippingValue: number) => {
+  return totalizers.map(totalizer => {
+    if (totalizer.id === shippingId) {
+      totalizer.value = newShippingValue
+    }
+    return totalizer
+  })
+}
+
+const getShipping = (totalizers: Totalizer[]) => {
+  return totalizers.find(totalizer => {
+    return totalizer.id === shippingId
+  })
+}
+
+const findDeliveryOptionById = (
+  deliveryOptions: DeliveryOption[],
+  deliveryOptionId: string
+): DeliveryOption => {
+  return deliveryOptions.find(option => {
+    return option.id === deliveryOptionId
+  })!
 }
 
+export const OrderShippingProvider = compose(
+  graphql(EstimateShipping, { name: 'EstimateShipping' }),
+  graphql(SelectDeliveryOption, { name: 'SelectDeliveryOption' })
+)(
+  ({
+    children,
+    EstimateShipping,
+    SelectDeliveryOption,
+  }: OrderShippingProps) => {
+    const { enqueue, listen, useQueueStatus } = useOrderQueue()
+    const { orderForm, setOrderForm } = useOrderForm()
+
+    const isQueueBusy = useQueueStatus(listen)
```

Comment:
[I had to change the API a bit.](https://github.com/vtex-apps/order-manager/tree/feature/order-status#usequeuestatuslisten-listenfunction-reactmutablerefobjectqueueevent) `useQueueStatus` returns a `ref` whose `.current` contains a **string** instead of a boolean. This follows the practice of avoiding boolean values that the store framework team has been adopting.

Please test this by linking the `feature/order-status` branch of `OrderManager`.